src/game/input.c

```diff
@@ -17,9 +17,32 @@ static bool m_KeyConflict[INPUT_ROLE_NUMBER_OF] = { false };
 static bool m_BtnConflict[INPUT_ROLE_NUMBER_OF] = { false };
 static int32_t m_HoldBack = 0;
 static int32_t m_HoldForward = 0;
+static int32_t m_HoldLook = 0;
 
+static void Input_CheckChangeTarget(INPUT_STATE *input);
 static INPUT_STATE Input_GetDebounced(INPUT_STATE input);
 
+static void Input_CheckChangeTarget(INPUT_STATE *input)
+{
+    if (g_Lara.gun_status != LGS_READY) {
+        return;
+    }
+
+    if (input->look) {
+        if (m_HoldLook >= 6)
+            m_HoldLook = 100;
+        else {
+            input->look = 0;
+            m_HoldLook++;
+        }
+    } else {
+        if (m_HoldLook && m_HoldLook != 100) {
+            input->change_target = 1;
+        }
+        m_HoldLook = 0;
+    }
+}
```

Comment:
Is this technique necessary, considering we use g_InputDB anyway, which is 1 only for the first frame the player holds the button? Or is it TR4+ behavior that it works with a slight delay?

Reply:
Ya all this code is based off TR4. (I did split up the gun new target and change target functions tho to account for all 3 modes).

Basically when Lara has guns equipped, this allows tapping look to change targets and only goes into look mode if the player holds down look.